src/Compilers/Server/VBCSCompiler/MetadataCache.cs

```diff
@@ -48,7 +46,7 @@ private ImmutableArray<ModuleMetadata> GetAllModules(ModuleMetadata manifestModu
                     moduleBuilder.Add(manifestModule);
                 }
 
-                var module = CreateModuleMetadata(PathUtilities.CombineAbsoluteAndRelativePaths(assemblyDir, moduleName), prefetchEntireImage: false);
+                var module = CreateModuleMetadata(PathUtilities.CombineAbsoluteAndRelativePaths(assemblyDir, moduleName)!, prefetchEntireImage: false);
```

Comment:
Why does this need a suppression?

Reply:
The parameter requires `string` and there is no way to annotate `CombineAbsoluteAndRelativePath` to return a non-null value. Can't even use `NotNullIfNotNull` here because it will return `null` based on the type of path in the second parameter.